R/hiper_glm.R

```diff
@@ -1,17 +1,47 @@
 #' @export
-hiper_glm <- function(design, outcome, model="linear") {
+hiper_glm <- function(design, outcome, model="linear", option = list()) {
 
   supported_model <- c("linear", "logit")
 
   if (!(model %in% supported_model)) {
     stop(sprintf("The model %s is not supported.", model))
   }
 
-  warning("`hiper_glm` is yet to be implemented.")
+  coef_estimate <- NULL
+
+  if (!is.null(option[["mle_solver"]])) {
+
+    if (option[["mle_solver"]] == "BFGS") {
+      # message("Using BFGS\n")
+
+      op <- stats::optim(par = rnorm(dim(design)[2]),
+                         fn = function(par) log_likelihood_linear(par, design, outcome),
+                         gr = function(par) log_likelihood_gradient_linear(par, design, outcome),
+                         control = list(fnscale=-1))
+
+      coef_estimate <- op$par
+
+    } else {
+      warning("No current plans for adding solver options other than pseudo-inverse and BFGS.")
+      stop()
```

Comment:
```suggestion
      stop("No current plans for adding solver options other than pseudo-inverse and BFGS.")
```
-1

Don't need to warn then stop, just stop.

Would also just make the message simpler, like "method ... not supported, please use  BFGS or pseudo inverse."